shared/api/instance.go

```diff
@@ -374,3 +378,80 @@ type InstanceSource struct {
 	// API extension: instance_allow_inconsistent_copy
 	AllowInconsistent bool `json:"allow_inconsistent" yaml:"allow_inconsistent"`
 }
+
+// InstanceEFI represents the EFI vars of a LXD virtual machine instance.
+//
+// swagger:model
+//
+// API extension: instances_efi_vars.
+type InstanceEFI struct {
+	// EFI variables map
+	// Hashmap key format is <efi-variable-name>-<UUID>
+	// Example: AmdAcpiVar-79941ecd-ed36-49d0-8124-e4c31ac75cd4
+	Variables map[string]*InstanceEFIVariable `json:"variables" yaml:"variables"`
+}
+
+// InstanceEFIVariable represents a EFI variable entry
+//
+// swagger:model
+//
+// API extension: instances_efi_vars.
+type InstanceEFIVariable struct {
+	// EFI variable data (HEX-encoded)
+	Data string `json:"data" yaml:"data"`
+
+	// EFI variable attributes
+	Attr uint32 `json:"attr" yaml:"attr"`
+
+	// EFI variable timestamp (HEX-encoded)
+	Timestamp string `json:"timestamp" yaml:"timestamp"`
+
+	// EFI variable digest (HEX-encoded)
+	Digest string `json:"digest" yaml:"digest"`
+}
+
+// CheckValid checks whether the InstanceEFI structure is valid.
+//
+// API extension: instances_efi_vars.
+func (e InstanceEFI) CheckValid() error {
+	for k, v := range e.Variables {
+		// Hashmap key format is <Var name>-<UUID>
+		// UUID length is 36
+		// Var name length is at least 1
+		// and we have "-" as a separator between name and UUID
+		if len(k) < 36+1+1 {
+			return fmt.Errorf("Bad EFI variable key: %q", k)
+		}
+
+		guid := k[len(k)-36:]
+		if _, err := uuid.Parse(guid); err != nil {
+			return fmt.Errorf("Bad EFI variable key: %q. Bad UUID", k)
```

Comment:
In general if we are getting an `error` from an action we should include it in the returned error, wrapped using `%w` so that the caller than inspect/assert on the wrapped error if needed (as well as it potentially offering additional useful info to the end user).

Reply:
https://go.dev/blog/go1.13-errors

Reply:
Have fixed! Thanks!